packages/itwin/tree-widget/src/components/trees/stateless/common/UseNodeHighlighting.tsx

```diff
@@ -0,0 +1,204 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect, useRef } from "react";
+import { isPresentationHierarchyNode } from "@itwin/presentation-hierarchies-react";
+
+import type { PresentationHierarchyNode, PresentationTreeNode } from "@itwin/presentation-hierarchies-react";
+
+interface UseNodeHighlightingProps {
+  rootNodes: PresentationTreeNode[] | undefined;
+  textToHighlight?: string;
+  caseSensitive?: boolean;
+  activeMatchIndex?: number;
+  onHighlightChanged?: (activeMatchIndex: number, matches: number) => void;
+}
+
+interface HighlightedChunk {
+  start: number;
+  end: number;
+}
+
+interface NodeChunkInfo {
+  chunks: HighlightedChunk[];
+  activeChunkIndex?: number;
+}
+
+interface NodeHighlightInfo {
+  startIndex: number;
+  matches: HighlightedChunk[];
+}
+
+interface HighlightState {
+  nodeInfoMap: Map<string, NodeHighlightInfo>;
+  totalMatches: number;
+}
+
+interface UseNodeHighlightingResult {
+  getLabel: (node: PresentationHierarchyNode) => React.ReactElement;
+}
+
+/** @internal */
+export function useNodeHighlighting({
+  rootNodes,
+  textToHighlight,
+  caseSensitive,
```

Comment:
I think we will always want to make case insensitive match.

Reply:
Removed the prop.